App/src/Repository/ExpenseRepository.php

```diff
@@ -12,4 +12,13 @@ public function __construct(ManagerRegistry $registry)
     {
         parent::__construct($registry, Expense::class);
     }
+
+    public function getExpensesOfGroupById(int $id): array
+    {
+        return $this->createQueryBuilder('expense')
+            ->where('expense.group = :group')
+            ->setParameter('group', $id)
+            ->getQuery()
+            ->getResult();
+    }
 }
```

Comment:
Pour respecter la nomenclature des autres repositories, je te suggère de garder les méthodes pour récupérer des entités dans le style `find...` pour une collection et `findOne...` pour une entité seule.

Note aussi que le nom de ton argument doit aider à comprendre rapidement ce que l'on attend. `int $id` nous indique que nous attendons un `$id` mais cela peu être un peu léger. Avec `$groupId` on comprend tout de suite que la fonction a besoin de l'ID d'un groupe et rien d'autre.

```suggestion
    public function findByGroupId(int $groupId): array
    {
        return $this->createQueryBuilder('expense')
            ->where('expense.group = :groupId')
            ->setParameter('groupId', $groupId)
            ->getQuery()
            ->getResult();
    }
}
```

Reply:
done